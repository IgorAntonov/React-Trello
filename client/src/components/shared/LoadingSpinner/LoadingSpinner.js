import React from 'react';
import styled, { keyframes } from 'styled-components';

export const LoadingSpinner = () => (
  <Svg width="100px" height="100px" viewBox="0 0 66 66">
    <Circle fill="none" strokeWidth="6" strokeLinecap="round" cx="33" cy="33" r="30" />
  </Svg>
);

export const PageSpinner = () => (
  <Wrapper>
    <LoadingSpinner />
  </Wrapper>
);

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 100vh;
`;

const rotator = keyframes`
  0% { transform: rotate(0deg); }
  100% { transform: rotate(270deg); }
`;
const colors = keyframes`
  0% { stroke: #4285F4; }
  25% { stroke: #DE3E35; }
  50% { stroke: #F7C223; }
  75% { stroke: #1B9A59; }
  100% { stroke: #4285F4; }
`;
const dash = keyframes`
  0% { stroke-dashoffset: 187; }
  50% {
    stroke-dashoffset: 47;
    transform:rotate(135deg);
  }
  100% {
    stroke-dashoffset: 187;
    transform:rotate(450deg);
  }
`;

export const Svg = styled.svg`
  animation: ${rotator} 1.4s linear infinite;
`;

const Circle = styled.circle`
  stroke-dasharray: 187;
  stroke-dashoffset: 0;
  transform-origin: center;
  animation:
    ${dash} 1.4s ease-in-out infinite, 
    ${colors} 5.6s ease-in-out infinite;
`;
