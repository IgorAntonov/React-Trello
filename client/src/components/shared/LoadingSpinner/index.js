export { LoadingSpinner, PageSpinner } from './LoadingSpinner';
