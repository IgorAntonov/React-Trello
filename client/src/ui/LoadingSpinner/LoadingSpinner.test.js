import React from 'react';
import { shallow } from 'enzyme';

import { LoadingSpinner, Svg } from './LoadingSpinner';

test(' <LoadingSpinner /> renders', () => {
  const wrapper = shallow(<LoadingSpinner />);
  expect(wrapper.exists()).toBe(true);
  expect(wrapper.find(Svg)).toHaveLength(1);
});
