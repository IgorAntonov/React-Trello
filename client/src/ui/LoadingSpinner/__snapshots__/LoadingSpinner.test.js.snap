// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` <LoadingSpinner/> should match snapshot 1`] = `
<LoadingSpinner__Svg
  height="100px"
  viewBox="0 0 66 66"
  width="100px"
>
  <LoadingSpinner__Circle
    cx="33"
    cy="33"
    fill="none"
    r="30"
    strokeLinecap="round"
    strokeWidth="6"
  />
</LoadingSpinner__Svg>
`;
